libcudacxx/include/cuda/std/__cccl/builtin.h

```diff
@@ -361,6 +361,10 @@
 #  define _CCCL_BUILTIN_PREFETCH(...)
 #endif // _CCCL_CHECK_BUILTIN(builtin_prefetch)
 
+#if (_CCCL_CHECK_BUILTIN(__builtin_add_overflow) || _CCCL_COMPILER(GCC)) && !_CCCL_DEVICE_COMPILATION()
```

Comment:
```suggestion
#if (_CCCL_CHECK_BUILTIN(__builtin_add_overflow) || _CCCL_COMPILER(GCC)) && !(_CCCL_CUDA_COMPILER(NVCC) && _CCCL_DEVICE_COMPILATION())
```
both nvc++ and clang-cuda support overflow builtins in host and device code :)

Reply:
I could check but from my experience builtin functions are far from our implementation

Reply:
You mean the implementation is worse?

Reply:
yes, they don't exploit anything internal contrary to their host version. I would expect a simple implementation on their side but makes sense to check

Reply:
I did a couple of experiments. Clang-19 `__builtin_add_overflow` generates suboptimal code. Two examples: 
`unsigned + unsigned -> int` uses `add.s64`, `unsigned + unsigned -> unsigned` plain code (not PTX carry-out).
I would use intrinsics only for host-code.